packages/api/src/router/event.ts

```diff
@@ -28,50 +27,50 @@ export const eventRouter = router({
             },
           },
           include: {
-            account: {
-              include: {
-                color: true,
-              },
-            },
+            account: true,
             attendees: true,
           },
         }),
       ]);
 
       // Get calendar Events
+
       const calendarEvents = await Promise.all(
         accounts.map(async (account) => {
           const service = getCalendarService(account);
           const { events } = await service.getEvents(input);
           return events.map((e) => ({
             ...e,
             account,
-            color: account.color.color,
           }));
         }),
       );
 
       // Merge events
       const allEvents = [...userEvents, ...calendarEvents.flat()];
 
-      // Remove duplicate events
-      const addedEvents = new Map<string | undefined, boolean>();
-      const newEvents = allEvents.filter((event) => {
-        const { id, startDate, endDate, title } = event;
-        // TODO: Find a better way for composite key
-        // This composite key is used to avoid duplicate events which have the same name and date, but a different id
-        // Note: Having different ids occur when a user with multiple accounts creates an event. The reason for this is that
-        // we cannot update non-primary calender ids only by having them attendees
-        const compositeKey = `${title},${startDate},${endDate}`;
-        // If an event with the same is added before, just skip it to avoid duplicates
-        if (addedEvents.has(compositeKey) || addedEvents.has(id)) {
-          return false;
+      // Deduplicate allEvents based on event.title, event.startDate and event.endDate
+      const dedupedEvents = allEvents.reduce((acc, event) => {
+        const existingEvent = acc.find(
+          (e) =>
+            e.title === event.title &&
+            e.startDate?.getTime() === event.startDate?.getTime() &&
+            e.endDate?.getTime() === event.endDate?.getTime(),
+        );
+        if (existingEvent) {
+          // If the event already exists, merge the attendees
+          existingEvent.attendees = [
+            ...(existingEvent.attendees || []),
+            ...(event.attendees || []),
+          ];
+        } else {
+          // If the event doesn't exist, add it to the accumulator
+          acc.push(event);
         }
-        addedEvents.set(compositeKey, true);
-        addedEvents.set(id, true);
-        return true;
-      });
-      return newEvents;
+        return acc;
+      }, [] as typeof allEvents[number][]);
+
+      return dedupedEvents;
```

Comment:
why custom reduce for this?

data structures like Map/Set are made for deduping and will perform better here with constant lookup times

Reply:
TBH there was no reason.
I just had this bug initially that displayed a single event for me.

This piece of code was added by github co-pilot and made it work for me. Now, given that you have fixed the old approach, I think I can revert it back to the `Map`/`Set` -- I agree that it's a good usage for them.